immuneML/reports/encoding_reports/FeatureDistributionReport.py

```diff
@@ -0,0 +1,173 @@
+import warnings
+from pathlib import Path
+
+import plotly.express as px
+
+from immuneML.analysis.data_manipulation.DataReshaper import DataReshaper
+from immuneML.data_model.dataset.RepertoireDataset import RepertoireDataset
+from immuneML.reports.ReportOutput import ReportOutput
+from immuneML.reports.ReportResult import ReportResult
+from immuneML.reports.encoding_reports.EncodingReport import EncodingReport
+from immuneML.util.PathBuilder import PathBuilder
+
+
+
+class FeatureDistributionReport(EncodingReport):
+    """
+    Plots a boxplot for each feature of the encoded dataset in one of the two modes:
+    in the 'normal' mode there are normal boxplots corresponding to each column of the 
+    encoded dataset matrix; in the 'sparse' mode all zero cells are eliminated before 
+    passing the data to the boxplots. If mode is set to 'auto', then it will automatically 
+    set to 'sparse' if the density of the matrix is below 0.01
+
+
+    Arguments:
+
+        grouping_label (str): The label name used for x-axis grouping of the barplots - defaults to "feature",
```

Comment:
Could we have one or more additional parameters that allows you to group the plots based on metadata? 
E.g., for featurevaluebarplot, there are color_grouping_label/row_grouping_label/column_grouping_label, which allow you to distinguish for example the features of diseased repertoires from features of healthy repertoires, by plotting them in different colors, row facets or column facets respectively. Probably color grouping is most intuitive.